src/services/raspberrydevconn/app.py

```diff
@@ -0,0 +1,255 @@
+            humidity, temperature = Adafruit_DHT.read_retry(self.sensor, self.pin)
+            if humidity is not None and temperature is not None:
+                self.logger.debug("Sensor connected correcly")
+            else:
+                self.logger.warning("Sensor not connected")
+            pass
+
+        if not self._onrpi:
+            self.logger.warning(
+                "Raspberry not found or error as occurred while Arduino init. Running as dummy device!"
+            )
+
+        # Initialize the thread for managing the read of the parameters
+        self._th = WIOThread(target=self._airhumidity, name="Air Humidity Thread")
+        self._th1 = WIOThread(
+            target=self._airtemperature, name="Air Temperature Thread"
+        )
+        self._th2 = WIOThread(
+            target=self._terrainhumidity, name="Terrain Humidity Thread"
+        )
+        upperRESTSrvcApp.subsribe_evt_stop(self._th.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th1.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th2.stop)
+
+        # Retrieve, by doing a GET request to the device config, the value of the sampleperiod of the parameters
+        self.wait_air_temp = (
+            self._catreq.reqREST(
+                "DeviceConfig", "/configs?path=/sensors/temp/sampleperiod"
+            ).json_response["v"]
+            / 1000
+        )
+        self.wait_air_hum = (
+            self._catreq.reqREST(
+                "DeviceConfig", "/configs?path=/sensors/airhum/sampleperiod"
+            ).json_response["v"]
+            / 1000
+        )
+        self.wait_soil_hum = (
+            self._catreq.reqREST(
+                "DeviceConfig", "/configs?path=/sensors/soilhum/sampleperiod"
+            ).json_response["v"]
+            / 1000
+        )
+
+        # Subscribe to mqtt topics where the data of the sampleperiod will be published and declare the function that will be executed when a value is published
+        self._catreq.subscribeMQTT("DeviceConfig", "/conf/sensors/temp/sampleperiod")
+        self._catreq.callbackOnTopic(
+            "DeviceConfig", "/conf/sensors/temp/sampleperiod", self.onMessageReceiveTemp
+        )
+        self._catreq.subscribeMQTT("DeviceConfig", "/conf/sensors/airhum/sampleperiod")
+        self._catreq.callbackOnTopic(
+            "DeviceConfig",
+            "/conf/sensors/airhum/sampleperiod",
+            self.onMessageReceiveAirhum,
+        )
+        self._catreq.subscribeMQTT("DeviceConfig", "/conf/sensors/soilhum/sampleperiod")
+        self._catreq.callbackOnTopic(
+            "DeviceConfig",
+            "/conf/sensors/soilhum/sampleperiod",
+            self.onMessageReceiveSoilhum,
+        )
+
+        # Start the threads
+        self._th.run()
+        self._th1.run()
+        self._th2.run()
+
+    # function of the threads
+    def _airhumidity(self):
+        while not self._th.is_stop_requested:
+            self.humidity = 50
```

Comment:
Maybe the default values can be put in the settings.json as well

Reply:
Fix in [7df4dd4](https://github.com/thegabriele97/WaterIoT/pull/25/commits/7df4dd4810af591f0ff404364eab6596534b1702)